runtime/cudaq/builder/kernel_builder.h

```diff
@@ -350,6 +351,10 @@ class kernel_builder : public details::kernel_builder_base {
   /// out of CUDA Quantum code
   std::unique_ptr<ExecutionEngine, void (*)(ExecutionEngine *)> jitEngine;
 
+  /// @brief Map created ExecutionEngines to a unique hash of the
+  /// ModuleOp they derive from.
+  std::map<ExecutionEngine *, std::size_t> jitEngineToModuleHash;
```

Comment:
Is there a reason we use a `std::map` here rather than a `std::unordered_map`?

Reply:
Nope, just what my fingers typed to start. I think it should be an unordered map, will make the change.